enterprise/jmx-monitoring/src/main/java/io/crate/beans/QueryStats.java

```diff
@@ -84,49 +96,49 @@ void inc(long duration) {
 
     private final Supplier<Map<StatementType, Metric>> metricByStmtType;
 
-    private volatile long lastUpdateTsInMillis = System.currentTimeMillis();
-
     public QueryStats(JobsLogs jobsLogs) {
         metricByStmtType = Suppliers.memoizeWithExpiration(
-            () -> {
-                long currentTs = System.currentTimeMillis();
-                Map<StatementType, Metric> metricByCommand = createMetricsMap(jobsLogs.jobsLog(), currentTs, lastUpdateTsInMillis);
-                lastUpdateTsInMillis = currentTs;
-                return metricByCommand;
-            },
+            () -> createMetricsMap(jobsLogs.metrics(), System.currentTimeMillis()),
             1,
             TimeUnit.SECONDS
         );
     }
 
-    static Map<StatementType, Metric> createMetricsMap(Iterable<JobContextLog> logEntries, long currentTs, long lastUpdateTs) {
+    static Map<StatementType, Metric> createMetricsMap(Iterable<Metrics> metrics, long currentTs) {
         Map<StatementType, Metric> metricsByStmtType = new HashMap<>();
-        long elapsedSinceLastUpdateInMs = currentTs - lastUpdateTs;
 
-        Metric total = new Metric(0, elapsedSinceLastUpdateInMs);
-        for (JobContextLog logEntry : logEntries) {
-            if (logEntry.started() < lastUpdateTs || logEntry.started() > currentTs) {
+        // the ALL metric needs the largest interval measurements are available against so it can compute the global
+        // averages (the /sec metrics are affected by this)
+        long largestMeasurementInterval = -1;
+        Metric total = new Metric(0, 0, 0);
+        for (Metrics classifiedMetrics : metrics) {
+            Histogram histogram = classifiedMetrics.histogram();
+            if (classifiedMetrics.startTime() > currentTs) {
                 continue;
             }
-            long duration = logEntry.ended() - logEntry.started();
-            total.inc(duration);
-            metricsByStmtType.compute(classificationType(logEntry), (key, oldMetric) -> {
+            long sumOfDurations = classifiedMetrics.sumOfDurations();
+            long currentMeasurementIntervalInMs = Math.max(currentTs - classifiedMetrics.startTime(), 0);
```

Comment:
If I understand this approach correctly we'd now have the op/s being calculated over the life-time of a node? (Or starting from the first time a metric is being created). I'm not sure if that is useful, it will flatten out the op/s, so it won't be possible to see changes as good as if we had a 5m or 1m based calculations.

I think previously the calculation was based on the interval of the metrics being accessed.